features/storage/TESTS/blockdevice/general_block_device/main.cpp

```diff
@@ -342,10 +342,11 @@ void test_contiguous_erase_write_read()
     utest_printf("\ncontiguous_erase_size=%" PRIu64, contiguous_erase_size);
 
     // Determine starting address
-    bd_addr_t start_address = rand();       // low 32 bytes
-    start_address += (uint64_t)rand() << 32;// high 32 bytes
-    start_address *= erase_size; // aligned with erase_block
-    start_address %= block_device->size() - contiguous_erase_size; //  Must fit all data
+    bd_addr_t start_address = rand();            // low 32 bytes
+    start_address += (uint64_t)rand() << 32;     // high 32 bytes
+    start_address %= block_device->size() - contiguous_erase_size - erase_size; //  fit all data + alignment reserve
+    start_address += erase_size;                 // add alignment reserve
+    start_address -= start_address % erase_size; // align with erase_block
```

Comment:
When the DataFlash tests started to work I found out there was still an issue with the start_address alignment. The old style size limitation would break the alignment.